flaky TestStreamsImmediateReset

```
=== RUN   TestStreamsImmediateReset
    webtransport_test.go:257: 
        	Error Trace:	webtransport_test.go:257
        	Error:      	Received unexpected error:
        	            	Application error 0x105: received HTTP/3 frame on bidirectional stream
        	Test:       	TestStreamsImmediateReset
--- FAIL: TestStreamsImmediateReset (0.01s)

--- a/client.go
+++ b/client.go
@@ -61,6 +61,9 @@ func (d *Dialer) init() {
 			}
 			id, err := quicvarint.Read(quicvarint.NewReader(str))
 			if err != nil {
+				if isWebTransportError(err) {
+					return true, nil
+				}
 				return false, err
 			}
 			d.conns.AddStream(conn, str, sessionID(id))

--- a/server.go
+++ b/server.go
@@ -89,6 +89,9 @@ func (s *Server) init() error {
 		}
 		id, err := quicvarint.Read(quicvarint.NewReader(str))
 		if err != nil {
+			if isWebTransportError(err) {
+				return true, nil
+			}
 			return false, err
 		}
 		s.conns.AddStream(qconn, str, sessionID(id))

